src/pyasm/search/sql.py

```diff
@@ -510,9 +510,11 @@ def connect(my):
                     encoding = 'utf8'
                 if not charset:
                     charset = 'utf8'
```

Comment:
I would add a check to see if my.port is actually specified
if not my.port:
      my.port = 3306